bases/renku_data_services/data_api/main.py

```diff
@@ -124,6 +125,12 @@ async def setup_sentry(_: Sanic) -> None:
 
     app.register_middleware(validate_null_byte, "request")
 
+    @app.middleware("response")
+    async def handle_head(request: Request, response: BaseHTTPResponse) -> None:
+        """Make sure HEAD requests return an empty body."""
+        if request.method == "HEAD":
+            response.body = None
```

Comment:
Does this mean, the response has already been created and is now discarded before being rendered?

Reply:
yes. the response needs to be created in some cases as we return the etag in the headers (which gets calculated based on DB state) and a client getting the etag without having to receive the whole entity is one of the main use-cases for head requests in our case.

So it doesn't save on server processing, just unnecessary network traffic.

There's not really a better way to solve this with Sanic from what I could tell, other than implementing individual "HEAD" handlers for every endpoint we have.